### PR TITLE
feat: Add configuration option to scope application listing / getting to particular namespace

### PR DESCRIPTION
**Pull request by grahamalama**
In https://github.com/argoproj-labs/argocd-image-updater/pull/854, AIU made changes so that it scanned applications across all namespaces. This introduced a few regressions for some use cases, namely https://github.com/argoproj-labs/argocd-image-updater/issues/902.

In this PR, I introduce a config option so that users can optionally scope the K8s client to a specific namespace (as was the case before https://github.com/argoproj-labs/argocd-image-updater/pull/854). Since searching across all namespaces was recently introduced as the default, I opted to keep that the default.

### CODE REVIEW

**Comment by grahamalama** [comment]
I was unsure how to add tests for this change. I saw [these tests](https://github.com/argoproj-labs/argocd-image-updater/blob/61ea007f2eb8b245270a5fde7e758c994b8ccdc3/pkg/argocd/argocd_test.go#L1013), but I'm not sure how to set up something similar to assert that "if the client is configured with an `appNamespace`, it will _only_ search for that namespace". Maybe asserting the fake client was called with certain parameters? Otherwise I think we'd end up testing the mock.

**Comment by chengfang** [comment]
@grahamalama thanks for the contribution. Can you please fix the DCO check failure? Is this PR also intended to address issue https://github.com/argoproj-labs/argocd-image-updater/issues/1111 ?

**Comment by grahamalama** [comment]
@chengfang just pushed the DCO fix.

> Is this PR also intended to address issue https://github.com/argoproj-labs/argocd-image-updater/issues/1111?

It could be I suppose! For me, my goal was to fix https://github.com/argoproj-labs/argocd-image-updater/issues/902 specifically. But I think it'd address both.

**Comment by chengfang** [comment]
> I was unsure how to add tests for this change. I saw [these tests](https://github.com/argoproj-labs/argocd-image-updater/blob/61ea007f2eb8b245270a5fde7e758c994b8ccdc3/pkg/argocd/argocd_test.go#L1013), but I'm not sure how to set up something similar to assert that "if the client is configured with an `appNamespace`, it will _only_ search for that namespace". Maybe asserting the fake client was called with certain parameters? Otherwise I think we'd end up testing the mock.

can we add tests with non-nil appNamespace, just for some basic sanity check?

**Comment by chengfang** [comment]
@grahamalama can you also update docs/install/reference.md to document the new flag? (https://argocd-image-updater.readthedocs.io/en/latest/install/reference/

**Comment by chengfang** [comment]
@grahamalama can you also take a look at the comment on the related issue https://github.com/argoproj-labs/argocd-image-updater/issues/1111#issuecomment-2840828900. Although it covers a different issue (still related to this one), I think the comment about using listApp vs getApp is also revelant here. Some namespace may contain large number of apps and list operation in this namespace could still be a performance bottleneck. If we know image-updater is configured to 1 namespace, we should be able to just use getApp.

**Comment by grahamalama** [comment]
> @grahamalama can you also take a look at the comment on the related issue [#1111 (comment)](https://github.com/argoproj-labs/argocd-image-updater/issues/1111#issuecomment-2840828900). Although it covers a different issue (still related to this one), I think the comment about using listApp vs getApp is also revelant here. Some namespace may contain large number of apps and list operation in this namespace could still be a performance bottleneck. If we know image-updater is configured to 1 namespace, we should be able to just use getApp.

Commented in #1111, and depending on the outcome of that conversation I'll update this PR accordingly. I will say though if the decision is to add some sort of caching option, I think that would be better suited for a separate PR.

**Comment by chengfang** [comment]
@grahamalama from the discussion here and in #1111, we are in agreement to improve this PR by using getApp instead of listApp when single namespace is configured in image-updater.

**Comment by grahamalama** [comment]
@chengfang, made the changes we discussed in #1111 and added the requested documentation

File: pkg/argocd/argocd.go [hunk 1]
```diff
@@ -25,13 +25,21 @@ import (
...
-	// List all applications across all namespaces (using empty labelSelector)
-	appList, err := client.ListApplications(v1.NamespaceAll)
+	// List all applications across configured namespace or all namespaces (using empty labelSelector)
+	if *client.appNamespace != v1.NamespaceAll {
+		return client.kubeClient.ApplicationsClientset.ArgoprojV1alpha1().Applications(*client.appNamespace).Get(ctx, appName, v1.GetOptions{})
+	}
+	return client.getApplicationInAllNamespaces(ctx, appName)
+}
+
+func (client *k8sClient) getApplicationInAllNamespaces(ctx context.Context, appName string) (*v1alpha1.Application, error) {
```
Comment (chengfang) [inline pkg/argocd/argocd.go:41]: the param `ctx` is not used in this new function. I think we should remove it.

File: docs/install/reference.md [hunk 2]
```diff
@@ -26,6 +26,10 @@ Runs the Argo CD Image Updater, possibly in an endless loop.
 
 ### Flags
 
+**--application-namespace *namespace***
+
+Specifies the Kubernetes namespace in which Argo Image Updater will manage Argo CD Applications when using the Kubernetes-based Application API. By default, applications in all namespaces are considered. This flag can be used to limit scope to a single namespace for performance, security, or organizational reasons.
```
Comment (chengfang) [inline docs/install/reference.md:31]:
```suggestion
Specifies the Kubernetes namespace in which Argo CD Image Updater will manage Argo CD Applications when using the Kubernetes-based Application API. By default, applications in all namespaces are considered. This flag can be used to limit scope to a single namespace for performance, security, or organizational reasons.
```

Comment (grahamalama) [reply docs/install/reference.md]: [3acfb8d](https://github.com/argoproj-labs/argocd-image-updater/pull/1112/commits/3acfb8d7a7558d7cf3a6874a87a8d9a0292f15fb)

File: pkg/argocd/argocd.go (hunk 1 above)
Comment (grahamalama) [reply pkg/argocd/argocd.go]: [eab19bd](https://github.com/argoproj-labs/argocd-image-updater/pull/1112/commits/eab19bdef06599e643a5f0613c537bf50c6c88ff)

**Review by chengfang — APPROVED** [review]
